spec/minefield_spec.rb

```diff
@@ -0,0 +1,282 @@
+      expect(m.field[0][0]).not_to equal m.field[5][5]
+      expect(m.field[0][9]).not_to equal m.field[9][0]
+      expect(m.field[3][7]).not_to equal m.field[7][3]
+    end
+  end
+
+  describe 'private methods' do
+    it '#generate_mines should be private' do
+      expect do
+        m.generate_mines
+      end.to raise_error
+    end
+    it '#generate_adjacent_mine_counts should be private' do
+      expect do
+        m.generate_adjacent_mine_counts
+      end.to raise_error
+    end
+  end
+
+  describe 'take_turn' do
+    let(:toy){ ToyMinefield.new }
+
+    before(:each) do
+      toy.field[0][0].place_mine
+      toy.generate_adjacent_mine_counts
+    end
+
+    context '#clear' do
+      it 'clears a cell with an adjacent mine' do
+        toy.take_turn({row: 0, column: 1, action: 'C'})
+        expect(toy.field[0][1].cleared).to be true
+      end
+
+      it 'explodes a cell with a mine' do
+        toy.take_turn({row: 0, column: 0, action: 'C'})
+        expect(toy.field[0][0].exploded?).to be true
+      end
+
+      it 'clears many cells if there are no adjacent mines' do
+        toy.take_turn({row: 0, column: 2, action: 'C'})
+        expect(toy.field[0][0].cleared).to be false
+        expect(toy.field[0][1].cleared).to be true
+        expect(toy.field[0][2].cleared).to be true
+        expect(toy.field[1][0].cleared).to be true
+        expect(toy.field[1][1].cleared).to be true
+        expect(toy.field[1][2].cleared).to be true
+        expect(toy.field[2][0].cleared).to be true
+        expect(toy.field[2][1].cleared).to be true
+        expect(toy.field[2][2].cleared).to be true
+      end
+    end
+
+    context '#auto_clear' do
+      it 'auto-clears surrounding cells if flags match' do
+        toy.field[0][0].flag
+        toy.field[0][1].clear
+        toy.take_turn({action: "A", row: 0, column: 1})
+        expect(toy.field[0][0].cleared).to be false
+        expect(toy.field[0][1].cleared).to be true
+        expect(toy.field[0][2].cleared).to be true
+        expect(toy.field[1][0].cleared).to be true
+        expect(toy.field[1][1].cleared).to be true
+        expect(toy.field[1][2].cleared).to be true
+        expect(toy.field[2][0].cleared).to be true
+        expect(toy.field[2][1].cleared).to be true
+        expect(toy.field[2][2].cleared).to be true
+      end
+      it 'will not auto-clear surrounding cells if flags don\'t match' do
+        toy.field[0][1].clear
+        toy.take_turn({action: "A", row: 0, column: 1})
+        expect(toy.field[0][0].cleared).to be false
+        expect(toy.field[0][1].cleared).to be true
+        expect(toy.field[0][2].cleared).to be false
+        expect(toy.field[1][0].cleared).to be false
+        expect(toy.field[1][1].cleared).to be false
+        expect(toy.field[1][2].cleared).to be false
+        expect(toy.field[2][0].cleared).to be false
+        expect(toy.field[2][1].cleared).to be false
+        expect(toy.field[2][2].cleared).to be false
```

Comment:
I think it's a judgment call whether you want to have this many expect lines or refactor the lines. On the one hands it's a lot of content, on the other hand it will be robust in letting you know if anything at all breaks. 

However, since your "it" block contains 8 "expect"s, which I haven't used multiple expects, if one line breaks does it just break the "it" block altogether? What I'm getting at here is I'm wondering if it will yield back to the user which line/cell broke.